commit ca5c83a012cc91ab8d38ebb416e5369619a830ca
Author: zoff99 <zoff99@users.noreply.github.com>
Date:   Fri Apr 22 19:50:58 2022 +0200

    0006_zoff_tc___custom_tweaks_with_global_vars

diff --git a/toxcore/tox.c b/toxcore/tox.c
index 2a97a3f19..4020e575e 100644
--- a/toxcore/tox.c
+++ b/toxcore/tox.c
@@ -30,6 +30,11 @@
 
 #include "../toxencryptsave/defines.h"
 
+bool global_filetransfer_is_resumable = false;
+bool global_force_udp_only_mode = false;
+bool global_do_not_sync_av = false;
+bool global_onion_active = true;
+
 #define SET_ERROR_PARAMETER(param, x) \
     do {                              \
         if (param != nullptr) {       \
