commit 6534ed38127c6d29408c8d1208873aa4bc1c7805
Author: zoff99 <zoff99@users.noreply.github.com>
Date:   Fri Apr 22 19:51:26 2022 +0200

    0007_zoff_tc___custom_tweaks_for_toxav

diff --git a/toxcore/Messenger.h b/toxcore/Messenger.h
index 4d30f056d..86f2c2e9f 100644
--- a/toxcore/Messenger.h
+++ b/toxcore/Messenger.h
@@ -100,9 +100,9 @@ typedef struct Messenger_Options {
 #define TOX_CAPABILITY_NEXT_IMPLEMENTATION ((uint64_t)1) << 63
 /* hardcoded capabilities of this version/branch of toxcore */
 #ifdef TOX_CAPABILITIES_ACTIVE
-#define TOX_CAPABILITIES_CURRENT (uint64_t)(TOX_CAPABILITY_CAPABILITIES | TOX_CAPABILITY_MSGV2 | TOX_CAPABILITY_MSGV3)
+#define TOX_CAPABILITIES_CURRENT (uint64_t)(TOX_CAPABILITY_CAPABILITIES | TOX_CAPABILITY_MSGV2 | TOX_CAPABILITY_MSGV3 | TOX_CAPABILITY_TOXAV_H264)
 #else
-#define TOX_CAPABILITIES_CURRENT (uint64_t)(TOX_CAPABILITY_CAPABILITIES)
+#define TOX_CAPABILITIES_CURRENT (uint64_t)(TOX_CAPABILITY_CAPABILITIES | TOX_CAPABILITY_TOXAV_H264)
 #endif
 /* size of the FLAGS in bytes */
 #define TOX_CAPABILITIES_SIZE sizeof(uint64_t)
diff --git a/toxcore/tox.h b/toxcore/tox.h
index 2261c1a59..7c9c4cc73 100644
--- a/toxcore/tox.h
+++ b/toxcore/tox.h
@@ -5556,6 +5556,22 @@ typedef Tox_Conference_Type TOX_CONFERENCE_TYPE;
 typedef enum Tox_File_Kind TOX_FILE_KIND;
 
 #endif
+
+/**
+ * Set the callback for the `friend_lossy_packet` event for a specific packet ID.
+ * to Pass NULL to unset.
+ * You need to set to NULL first, only then you are allowed to change it
+ *
+ */
+void tox_callback_friend_lossy_packet_per_pktid(Tox *tox, tox_friend_lossy_packet_cb *callback, uint8_t pktid);
+
+/**
+ * Set the callback for the `friend_lossless_packet` event for a specific packet ID.
+ * to Pass NULL to unset.
+ *
+ */
+void tox_callback_friend_lossless_packet_per_pktid(Tox *tox, tox_friend_lossless_packet_cb *callback, uint8_t pktid);
+
 //!TOKSTYLE+
 
 #endif // C_TOXCORE_TOXCORE_TOX_H
