commit e696d0508d66e77a251a6c1a8ee6ecaddb8dd2b6
Author: zoff99 <zoff99@users.noreply.github.com>
Date:   Fri Apr 22 19:55:11 2022 +0200

    0011_zoff_tc___onion_active

diff --git a/toxcore/Messenger.c b/toxcore/Messenger.c
index e36c4af55..56fcb4bbf 100644
--- a/toxcore/Messenger.c
+++ b/toxcore/Messenger.c
@@ -45,6 +45,7 @@
  */
 
 extern bool global_force_udp_only_mode;
+extern bool global_onion_active;
 
 static_assert(MAX_CONCURRENT_FILE_PIPES <= UINT8_MAX + 1,
               "uint8_t cannot represent all file transfer numbers");
@@ -2603,7 +2604,9 @@ void do_messenger(Messenger *m, void *userdata)
     }
 
     do_net_crypto(m->net_crypto, userdata);
-    do_onion_client(m->onion_c);
+    if (global_onion_active) {
+        do_onion_client(m->onion_c);
+    }
     do_friend_connections(m->fr_c, userdata);
     do_friends(m, userdata);
 #ifndef VANILLA_NACL
diff --git a/toxcore/tox.c b/toxcore/tox.c
index 6d7d5d3c0..54c58f75b 100644
--- a/toxcore/tox.c
+++ b/toxcore/tox.c
@@ -4950,3 +4950,8 @@ void tox_set_do_not_sync_av(bool value)
 {
     global_do_not_sync_av = value;
 }
+
+void tox_set_onion_active(bool value)
+{
+    global_onion_active = value;
+}
diff --git a/toxcore/tox.h b/toxcore/tox.h
index 4c95f9942..e7792ea40 100644
--- a/toxcore/tox.h
+++ b/toxcore/tox.h
@@ -386,6 +386,7 @@ uint32_t tox_max_hostname_length(void);
 
 void tox_set_force_udp_only_mode(bool value);
 void tox_set_do_not_sync_av(bool value);
+void tox_set_onion_active(bool value);
 
 /** @{
  * @name Global enumerations
