commit 6472fd002f4997718a2a3fab46bbd8f1d8d29dcc
Author: zoff99 <zoff99@users.noreply.github.com>
Date:   Mon Apr 18 09:43:14 2022 +0200

    onion active

diff --git a/toxcore/Messenger.c b/toxcore/Messenger.c
index bc79c37b7..9306d04f5 100644
--- a/toxcore/Messenger.c
+++ b/toxcore/Messenger.c
@@ -42,6 +42,7 @@
  */
 
 extern bool global_force_udp_only_mode;
+extern bool global_onion_active;
 
 static_assert(MAX_CONCURRENT_FILE_PIPES <= UINT8_MAX + 1,
               "uint8_t cannot represent all file transfer numbers");
@@ -2419,7 +2420,9 @@ void do_messenger(Messenger *m, void *userdata)
     }
 
     do_net_crypto(m->net_crypto, userdata);
-    do_onion_client(m->onion_c);
+    if (global_onion_active) {
+        do_onion_client(m->onion_c);
+    }
     do_friend_connections(m->fr_c, userdata);
     do_friends(m, userdata);
     m_connection_status_callback(m, userdata);
diff --git a/toxcore/tox.c b/toxcore/tox.c
index ee075cedd..f68f1d05b 100644
--- a/toxcore/tox.c
+++ b/toxcore/tox.c
@@ -3030,3 +3030,8 @@ void tox_set_do_not_sync_av(bool value)
     global_do_not_sync_av = value;
 }
 
+void tox_set_onion_active(bool value)
+{
+    global_onion_active = value;
+}
+
diff --git a/toxcore/tox.h b/toxcore/tox.h
index 5b4f435ef..36a03cc12 100644
--- a/toxcore/tox.h
+++ b/toxcore/tox.h
@@ -386,6 +386,7 @@ uint32_t tox_max_hostname_length(void);
 
 void tox_set_force_udp_only_mode(bool value);
 void tox_set_do_not_sync_av(bool value);
+void tox_set_onion_active(bool value);
 
 /** @{
  * @name Global enumerations
