commit 1309d202b0e4f1e4f79faf6869c66a2c658c2c00
Author: zoff99 <zoff99@users.noreply.github.com>
Date:   Sat May 21 15:05:20 2022 +0200

    0019_zoff_tc___fix_unlock_on_api_function

diff --git a/toxcore/tox.c b/toxcore/tox.c
index b58cca9de..c7b30d4d0 100644
--- a/toxcore/tox.c
+++ b/toxcore/tox.c
@@ -3927,8 +3927,8 @@ uint32_t tox_group_by_chat_id(const Tox *tox, const uint8_t *chat_id, Tox_Err_Gr
     if (chat == nullptr)
     {
         SET_ERROR_PARAMETER(error, TOX_ERR_GROUP_STATE_QUERIES_GROUP_NOT_FOUND);
-        return UINT32_MAX;
         tox_unlock(tox);
+        return UINT32_MAX;
     }
 
     uint32_t ret = chat->group_number;
